awesome_owl/static/src/counter/counter.js

```diff
@@ -0,0 +1,19 @@
+import { Component, useState } from "@odoo/owl";
+
+export class Counter extends Component {
+    static template = "awesome_owl.counter"
+    static props = {
+        onChange: { type: Function, optional: true }
+    }
+
+    setup() {
+        this.state = useState({ value: 0 });
+    }
+
+    increment() {
+        this.state.value++;
+        if (this.props.onChange) {
+            this.props.onChange();
+        }
```

Comment:
You can use the cool `?` in js


```suggestion
        this.props?.onChange();
```
this only runs the function if this.props exists